src/components/Todo/TodoTask.tsx

```diff
@@ -0,0 +1,174 @@
+import classNames from 'classnames';
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { useTodos } from '../Contexts/TodosContext';
+import { useLoadingTodos } from '../Contexts/LoadingTodosContext';
+import { deleteTodo, updateTodo } from '../../api/todos';
+import { useErrorMessage } from '../Contexts/ErrorMessageContext';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoTask: React.FC<Props> = ({
+  todo,
+}) => {
+  const { setTodos } = useTodos();
+  const { loadingTodos, setLoadingTodos } = useLoadingTodos();
+  const { handleShowError } = useErrorMessage();
+
+  const [editingTodoTitle, setEditingTodoTitle] = useState(todo.title);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const inputFieldRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputFieldRef.current) {
+      inputFieldRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleChangeTodoTitle = async () => {
+    const normalizedTitle = editingTodoTitle.trim();
+
+    if (normalizedTitle === todo.title.trim()) {
+      setIsEditing(false);
+
+      return;
+    }
+
+    setLoadingTodos(prev => [...prev, todo.id]);
+    if (normalizedTitle) {
+      try {
+        const updatedTodo = await updateTodo(todo.id,
+          { title: editingTodoTitle });
+
+        setTodos(prev => prev.map((currentTodo) => {
+          if (currentTodo.id === updatedTodo.id) {
+            return updatedTodo;
+          }
+
+          return currentTodo;
+        }));
+      } catch (error) {
+        handleShowError('Encounted error while trying to update Todo');
+      }
+    }
+
+    if (!normalizedTitle) {
+      try {
+        await deleteTodo(todo.id);
+        setTodos(prev => prev.filter(({ id }) => id !== todo.id));
+      } catch {
+        handleShowError('Encounted error while trying to delete Todo');
+      }
+    }
+
+    setLoadingTodos(prev => prev.filter(id => id !== todo.id));
+
+    setIsEditing(false);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    handleChangeTodoTitle();
+  };
+
+  const handleComplete = async (todoId: number,
+    event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
```suggestion
  const handleComplete = async (
    todoId: number,
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
```
Fix formatting in your code in all places